Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -631,13 +634,9 @@ - (void)writeObjectToDefaults:(NSString *)key value:(NSObject *)value {
 
 - (void)persistPrefsToDisk {
     @synchronized (self) {
-        NSDictionary *persistenceDict = [self.persistenceDict copy];
+        if (!self.persistenceDict) return;
```

Comment:
I didn't check the commit history.  But the `NSData *data = [NSKeyedArchiver archivedDataWithRootObject:self.persistenceDict];` on the next line effectively does a deep copy of the dict and it's data.

The `[self.persistenceDict copy]` copy doesn't necessarily do a deep copy which can lead to a crash later.